platform/darwin/src/MGLTypes.h

```diff
@@ -70,6 +70,21 @@ typedef NS_OPTIONS(NSUInteger, MGLMapDebugMaskOptions) {
 #endif
 };
 
+/**
+ A structure containing information about a transition.
+ */
+typedef struct MGLTransition {
+    /**
+     The duration in seconds to animate any changes to the style URL or to layout and paint attributes.
+     */
+    NSTimeInterval duration;
+    
+    /**
+     The delay in seconds before applying any changes to the style URL or to layout and paint attributes.
+     */
+    NSTimeInterval delay;
+} MGLTransition;
+
```

Comment:
We should provide a factory method MGLTransitionMake(). We might also want to add transition-related methods to NSValue(MGLAdditions) to make it easier to work with MGLTransition instances when an object is needed.